lib/chart/line/axis_svg.ex

```diff
@@ -9,82 +9,58 @@ defmodule Chart.Line.AxisSvg do
       settings
       |> Map.put(:view, View)
 
-    x_axis = View.get_axis(settings, {1, 0})
-    y_axis = View.get_axis(settings, {0, 1})
-
     ~E"""
     <g class="axis">
-      <% y_minor_ticks = settings[y_axis].minor_ticks %>
-      <defs>
-        <line id="y-minor-line-ticks" x1="<%= y_minor_ticks.length %>"/>
-      </defs>
-      <g class="y-minor-ticks">
-        <%= for py <- y_minor_ticks.positions do %>
-          <use xlink:href="#y-minor-line-ticks"
-            transform="<%= @view.translate_y_ticks(settings[y_axis].line, y_minor_ticks.length, y_minor_ticks.gap, py) %>" />
-        <% end %>
-      </g>
-
-      <% y_major_ticks = settings[y_axis].major_ticks %>
-      <defs>
-        <line id="y-major-line-ticks" x1="<%= y_major_ticks.length %>"/>
-      </defs>
-      <g class="y-major-ticks">
-        <%= for py <- y_major_ticks.positions do %>
-          <use xlink:href="#y-major-line-ticks"
-            transform="<%= @view.translate_y_ticks(settings[y_axis].line, y_major_ticks.length, y_major_ticks.gap, py) %>" />
-        <% end %>
-      </g>
-
-      <% x_minor_ticks = settings[x_axis].minor_ticks %>
-      <%= inspect x_minor_ticks %>
-
+    <%= for {axis, vector} <- @axis_table do %>
+      <% settings_ax = settings[axis] %>
+      <% minor_ticks = settings_ax.minor_ticks %>
       <defs>
-        <line id="x-minor-line-ticks" y1="<%= x_minor_ticks.length %>"/>
+        <%= case vector do %>
+          <% {1, 0} -> %><line id="<%= Atom.to_string(axis) %>-minor-line-ticks" y1="<%= minor_ticks.length %>"/>
+          <% {0, 1} -> %><line id="<%= Atom.to_string(axis) %>-minor-line-ticks" x1="<%= minor_ticks.length %>"/>
+        <% end%>
```

Comment:
nebylo by lepsi si to vyresit ve view pro axis?

Reply:
Oki zkusim.